src/emulator/auth/state.ts

```diff
@@ -550,6 +550,45 @@ export class ProjectState {
     }
   }
 }
+
+export class AgentProjectState extends ProjectState {
+  private tenantForTenantId: Map<string, Tenant> = new Map();
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.get is added
+  getTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.list is added
+  listTenants(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.create is added
+  createTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.patch is added
+  updateTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.delete is added
+  deleteTenant(): void {
+    return;
+  }
+}
+
+export class TenantProjectState extends ProjectState {
```

Comment:
A few things should have only one single instance that applies to both the agent + all tenants -- e.g. `oneAccountPerEmail`, `authCloudFunction` -- how would you like to handle this?

e.g. when `oneAccountPerEmail` is updated, it should affect the behavior of the agent project and all tenants under.

Reply:
Ah, thanks for pointing that out - I've updated `state.ts` such that the top level project settings are only set in the `AgentProjectState`s and `TenantProjectState`s simply refer to their parent state's settings. Thanks for your patience while I was debugging a gnarly [shadowing bug](https://stackoverflow.com/questions/53584705/javascript-extending-es6-class-setter-will-inheriting-getter#:~:text=the%20instance%20x%20will%20not,of%20the%20setter%2C%20as%20undefined.). LMK what you think with the new revision